pandas/io/parsers/readers.py

```diff
@@ -799,7 +799,7 @@ def read_fwf(
 
     Returns
     -------
-    DataFrame or TextParser
```

Comment:
Does `read_fwf` support `chunksize` or `iterator`? If so, it could return a TextFileReader.

Reply:
Yeah, tried this a few days ago and raised an error, but did something wrong.

Added the reader back in

Reply:
Thanks for checking

Reply:
Maybe in the future: Could also create overloads for the TextFileReader case. I think you would need to lie to mypy and declare `chunksize` and `iterator` as required keyword-arguemnts in the overload definitions.

Reply:
Yeah, wanted to deprecate passing non key word arguments here and add this after we can enforce this. Will put up a pr shortly